### DISCUSSION TITLE
Validation d'un panier suite à un payment

### DISCUSSION THREAD

**Discussion by saissaoui** [PayGreen/documentation#27] [Q&A] [answered]
Bonjour,
Je suis en train d'implémenter les payments via paygreen dans un site e-commerce, et le flow que j'ai mis en place est le suivant :
- Création du panier 
- Création d'un payment order via l'api paygreen
- Utilisation de la hosted page link du payment order crée
- Capture de l'event "payment_order.successed" via un webhook afin de pouvoir valider la commande
Ma question est la suivante: pour pouvoir valider la commande j'ai besoin de faire transiter l'information de l'id du panier ou de ma commande, quel est le meilleur moyen de le faire ( j'ai vu qu'il y avait un param metadata dans l'event capturé par le webhook mais il n y a aucun moyen d'envoyer de la metadata lors de la création du payment order)
Merci d'avance

**Comment by AdrienBr** [comment, marked as answer]
Bonjour @saissaoui ,
Je vous confirme que vous pouvez envoyer un attribut metadata lors de la création du Payment Order, par exemple : 

```
    "metadata": {
        "key": "value"
    }
```

Est-ce que cela solutionne votre problème ?

Cordialement,
Adrien

**Reply by saissaoui** [reply]
Bonjour @AdrienBr ,
Merci pour la réactivité, ça solutionne parfaitement mon problème.
Bonne journée
